airbyte-integrations/connectors/source-crowdstrike-falcon/source_crowdstrike_falcon/source.py

```diff
@@ -0,0 +1,275 @@
+    product_type_desc: str
+
+class behaviorModel(BaseModel):
+    tactic: str
+    technique: str
+
+class DetectModel(BaseModel):
+    created_timestamp: str
+    detection_id: str
+    device: DeviceModel
+    behaviors: List[behaviorModel]
+    first_behavior: str
+    last_behavior: str
+    max_severity_displayname: str
+    status: str
+    seconds_to_triaged: int
+    seconds_to_resolved: int
+    date_updated: str
+
+class CrowdstrikeFalcon(HttpStream, ABC):
+    limit = 10
+
+    def __init__(self, url_base: str,*args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def path(self, **kwargs) -> str:
+        return self.resource_path
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        data = response.json()
+        meta = data.get('meta')
+        
+        if meta:
+            pagination = meta.get('pagination')
+        
+        offset = int(pagination.get('offset'))
+        limit = int(pagination.get('limit'))
+        total = int(pagination.get('total'))
+        
+        offset += limit
+        
+        if offset < total:
+            return { "offset": offset, "limit": limit}
+
+        return None
+        
+
+    def request_params(self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None) -> MutableMapping[str, Any]:
+        curr_date = stream_state.get(self.cursor_field)
+        params = {
+            "limit": self.limit,
+            "sort": self.order_field,
+            "offset": 0
+        }
+
+        if next_page_token:
+            params.update(next_page_token)
+        
+        return params
+
+    def parse_response(self, response: requests.Response,stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        data = response.json()
+
+        ids = data.get('resources')
+
+        yield {
+            "ids": ids
+        }
+
+class IncrementalCrowdstrikeFalcon(CrowdstrikeFalcon, IncrementalMixin):
+    cursor_init_value = "1900-01-01T00:00:00.000Z"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._cursor_value = self.cursor_init_value
+        self._is_finished = False
+
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        return { self.cursor_field: self._cursor_value }
+
+    @state.setter
+    def state(self, value: MutableMapping[str, Any]):
+        self._cursor_value = value.get(self.cursor_field)
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        curr_date = stream_state.get(self.cursor_field)
+        params = super().request_params(stream_state=self.state, next_page_token=next_page_token, **kwargs)
+        curr_date = self.state.get(self.cursor_field)
+        filter_param = {"filter": self.filter_field + ":>'" + curr_date + "'"}
```

Comment:
the pythonic way of concatanting strings (or actually everything that can be converted to string) is f-string.

Example:
`f"{self.filter_field}:>'{curr_date}'"`

Reply:
It works as long as your variable has `__str__` method implemented or using other definition as long as `str(variable)` does not crash ;-)